### PR TITLE
CHOWTapeModel: 2.10.0 -> 2.11.4

### PR DESCRIPTION
**Pull request by polygon**
## Description of changes

* Update upstream version
* Use GCC11 since GCC12 is incompatible with JUCE version
* Release new CLAP version of the plugin
* Do not copy static code library, it is not required for operation and reduces the closure size by 10MB
* JUCE uses dlopen to load X11 libraries, add link flags so that these libraries are part of the rpath
* Replace `gcc-unwrapped` dependency with `stdenv.cc.cc`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
- [] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Tested plugins inside of Bitwig-Studio DAW
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/CHOWTapeModel/default.nix [hunk 1]
```diff
@@ -41,36 +76,51 @@ stdenv.mkDerivation rec {
     libsepol
     libsysprof-capture
     libthai
+    libuuid
     libxkbcommon
     lv2
     pcre
+    pcre2
```
Comment (magnetophon) [inline pkgs/applications/audio/CHOWTapeModel/default.nix:83]: Do we need both pcre and pcre2?

**Review by magnetophon — APPROVED** [review]
Result of `nixpkgs-review pr 268076` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>CHOWTapeModel</li>
  </ul>
</details>

File: pkgs/applications/audio/CHOWTapeModel/default.nix (hunk 1 above)
Comment (polygon) [reply pkgs/applications/audio/CHOWTapeModel/default.nix:83]: Good question, I went after what the configuration tool reported as missing. When I remove pcre2 it says:

```
Package libpcre2-8 was not found in the pkg-config search path.
Perhaps you should add the directory containing `libpcre2-8.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libpcre2-8', required by 'glib-2.0', not found
```

When I remove `pcre`, I get:

```
Package libpcre was not found in the pkg-config search path.
Perhaps you should add the directory containing `libpcre.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libpcre', required by 'libselinux', not found
```

I haven't checked if they are truly required or if the build succeeds without, though,

Comment (magnetophon) [reply pkgs/applications/audio/CHOWTapeModel/default.nix:83]: OK, thanks, looks like they are needed then.

Comment (magnetophon) [reply pkgs/applications/audio/CHOWTapeModel/default.nix:83]: OK, thanks, looks like they are needed then.

**Pull request merged**